test/remote/block_ip_tests.js

```diff
@@ -47,74 +40,77 @@ test(
   }
 )
 
+var client = restify.createJsonClient({
+  url: 'http://127.0.0.1:' + config.listen.port
+})
+
 test(
   'missing ip',
   function (t) {
-    return client.postAsync('/blockIp', {})
-      .then(function (req, res, obj) {
-        //missing parameters
-      }, function(err){
-        t.equal(err.statusCode, 400, 'bad request returns a 400')
-        t.type(err.restCode, 'string', 'bad request returns an error code')
-        t.type(err.message, 'string', 'bad request returns an error message')
+    client.post('/blockIp', {},
+      function (err, req, res, obj) {
+        t.equal(res.statusCode, 400, 'bad request returns a 400')
+        t.type(obj.code, 'string', 'bad request returns an error code')
+        t.type(obj.message, 'string', 'bad request returns an error message')
         t.end()
-      })
+      }
+    )
   }
 )
 
 test(
   'well-formed request',
   function (t) {
-    return client.postAsync('/check', { email: TEST_EMAIL, ip: TEST_IP, action: 'accountLogin' })
-      .spread(function (req, res, obj) {
+    client.post('/check', { email: TEST_EMAIL, ip: TEST_IP, action: 'accountLogin' },
+      function (err, req, res, obj) {
         t.equal(res.statusCode, 200, 'check worked')
         t.equal(obj.block, false, 'request was not blocked')
 
-        return client.postAsync('/blockIp', { ip: TEST_IP })
-      })
-      .spread(function (req, res, obj) {
-        t.equal(res.statusCode, 200, 'block request returns a 200')
-        t.ok(obj, 'got an obj, make jshint happy')
+        client.post('/blockIp', { ip: TEST_IP },
+          function (err, req, res, obj) {
+            t.notOk(err, 'block request is successful')
+            t.equal(res.statusCode, 200, 'block request returns a 200')
+            t.ok(obj, 'got an obj, make jshint happy')
 
-        return client.postAsync('/check', { email: TEST_EMAIL, ip: TEST_IP, action: 'accountLogin' })
-      })
-      .spread(function (req, res, obj) {
-        t.equal(res.statusCode, 200, 'check worked')
-        t.equal(obj.block, true, 'request was blocked')
-        t.end()
-      })
-      .catch(function(err){
-        t.fail(err)
-        t.end()
-      })
+            client.post('/check', { email: TEST_EMAIL, ip: TEST_IP, action: 'accountLogin' },
+              function (err, req, res, obj) {
+                t.equal(res.statusCode, 200, 'check worked')
+                t.equal(obj.block, true, 'request was blocked')
+                t.end()
+              }
+            )
+          }
+        )
+      }
+    )
```

Comment:
Hrm, looks like this undid some of @larissagaulia's fine work on promisifying the tests from https://github.com/mozilla/fxa-customs-server/commit/6fadc52bfabfc3d08dbbdf3f4a42bf67ec9b54cf, was this a bad merge of some sort @seanmonstar?